# This file is part of HiGFXback

# requires
REQUIRES="cmake3-build gcc-g++-build gdk-pixbuf-build libaom-build libde265-build libwebp-build x265-build"

PKG_CONFIG_PATH=/x11/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
GENERATOR=${GENERATOR-Unix Makefiles}
PKG_CONFIG_PATH=/x11/lib/pkgconfig cmake -DCMAKE_BUILD_WITH_INSTALL_RPATH=ON -DCMAKE_INSTALL_PREFIX=/x11 -DCMAKE_INSTALL_RPATH=/x11/lib -B build -G "$GENERATOR"

# build
cmake --build build

# install
install -d $DESTDIR/x11/bin
install build/examples/heif-convert $DESTDIR/x11/bin
install build/examples/heif-enc $DESTDIR/x11/bin
install build/examples/heif-info $DESTDIR/x11/bin
install build/examples/heif-thumbnailer $DESTDIR/x11/bin
install -d $DESTDIR/x11/include/libheif
install -m 644 build/libheif/heif_version.h $DESTDIR/x11/include/libheif
install -m 644 libheif/heif.h $DESTDIR/x11/include/libheif
install -d $DESTDIR/x11/lib
install build/libheif/libheif.so.1.* $DESTDIR/x11/lib/libheif.so.1
ln -sf libheif.so.1 $DESTDIR/x11/lib/libheif.so
install -d $DESTDIR/x11/lib/gdk-pixbuf-2.0/2.10.0/loaders
install build/gdk-pixbuf/libpixbufloader-heif.so $DESTDIR/x11/lib/gdk-pixbuf-2.0/2.10.0/loaders
install -d $DESTDIR/x11/lib/pkgconfig
install -m 644 build/libheif.pc $DESTDIR/x11/lib/pkgconfig
test $DESTDIR || /x11/bin/gdk-pixbuf-query-loaders > /x11/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache

# build.pc
install -d $DESTDIR/x11/share/pkgconfig
cat > $DESTDIR/x11/share/pkgconfig/libheif-build.pc << EOF
Name: libheif
Version: 1.16.2
Description: High Efficiency Image Format (HEIF) library
Requires: $REQUIRES

devel=\\
/x11/include/libheif/heif.h \\
/x11/include/libheif/heif_version.h \\
/x11/lib/libheif.so \\
/x11/lib/pkgconfig/libheif.pc

exec=\\
/x11/bin/heif-convert \\
/x11/bin/heif-enc \\
/x11/bin/heif-info \\
/x11/bin/heif-thumbnailer \\
/x11/lib/libheif.so.1 \\
/x11/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-heif.so
EOF

exit
--- libheif-1.16.2.orig/gdk-pixbuf/CMakeLists.txt
+++ libheif-1.16.2/gdk-pixbuf/CMakeLists.txt
@@ -3,10 +3,7 @@
   pkg_check_modules(GDKPIXBUF2 gdk-pixbuf-2.0)
 
   if(GDKPIXBUF2_FOUND)
-    execute_process(
-      COMMAND ${PKG_CONFIG_EXECUTABLE} gdk-pixbuf-2.0 --variable gdk_pixbuf_moduledir
-              --define-variable=prefix=${CMAKE_INSTALL_PREFIX} OUTPUT_VARIABLE GDKPIXBUF2_MODULE_DIR
-      OUTPUT_STRIP_TRAILING_WHITESPACE)
+    pkg_get_variable(GDKPIXBUF2_MODULE_DIR gdk-pixbuf-2.0 gdk_pixbuf_moduledir)
 
     add_library(pixbufloader-heif MODULE pixbufloader-heif.c)
 
